### PR TITLE
 Update all targets to at least java7 and android14

### PR DESCRIPTION
**Pull request by bogdandrutu**
gRPC updated to java7/android14 for io.context and all their artifacts see https://github.com/grpc/grpc-java/pull/4727

Currently we still cannot use try-with-resources (android 19) or ThreadLocalRandom (android 21).

### CODE REVIEW

**Comment by bogdandrutu** [comment]
We would like to have the exporters work on Android as well. Preferably we should have the SD monitoring as well but currently uses "java.lang.management.RuntimeMXBean"

File: api/build.gradle [hunk 1]
```diff
@@ -6,7 +6,8 @@ dependencies {
 
     compileOnly libraries.auto_value
 
-    signature "org.codehaus.mojo.signature:java16:+@signature"
+    signature "org.codehaus.mojo.signature:java17:+@signature"
+    signature "net.sf.androidscents.signature:android-api-level-14:+@signature"
```
Comment (adriancole) [inline api/build.gradle]: neat. I didn't know you can check for android!

**Comment by adriancole** [comment]
ps to make justification clear this isnt just because of grpc, it is because we depend on grpc context which as a part of grpc moves the floor version to same as all of grpc.

is this summary right?

File: api/build.gradle (hunk 1 above)
Comment (zhangkun83) [inline api/build.gradle]: gRPC uses specific version numbers `1.0` and `4.0_r4`, while you use `+` here. Any reason to be different?

Comment (bogdandrutu) [reply api/build.gradle]: fixed.

**Comment by ejona86** [comment]
grpc/grpc-java#4727 didn't entirely drop our support of Java 6, since source and target remained 1.6. I just merged grpc/grpc-java#4799 which updated the source and target for the artifacts we were able to. A few tooling compatibility issues (grpc-java build errors) arose for some artifacts which we'll handle separately.

**Pull request merged**